lib/logstash/filters/cidr.rb

```diff
@@ -147,18 +147,18 @@ def filter(event)
             IPAddr.new(n)
           end
         rescue ArgumentError => e
-          @logger.warn("Invalid IP network, skipping", :network => n, :event => event)
+          @logger.warn("Invalid IP network, skipping", :network => n, :event => event.to_hash)
           nil
         end
       end
 
     else #networks come from array in config file
 
-      network = @network.collect do |n|
+      network = @network.map {|nw| event.sprintf(nw) }.map {|nw| nw.split(",") }.flatten.collect do |n|
         begin
-          IPAddr.new(event.sprintf(n))
+          IPAddr.new(n.strip)
         rescue ArgumentError => e
-          @logger.warn("Invalid IP network, skipping", :network => n, :event => event)
+          @logger.warn("Invalid IP network, skipping", :network => n, :event => event.to_hash)
```

Comment:
Nit: By doing the stripping on line 159 instead of on line 157 we're logging the non-stripped string which might give the user the wrong impression that leading or trailing whitespace is the culprit.